src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1025,7 +1025,18 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                 var current = context.LastDatabaseChangeVector ?? DocumentsStorage.GetDatabaseChangeVector(context);
                 var conflictStatus = ChangeVectorUtils.GetConflictStatus(_changeVector, current);
                 if (conflictStatus != ConflictStatus.Update)
+                {
+                    if (string.IsNullOrEmpty(_connectionInfo.SourceDatabaseBase64Id) == false)
+                    {
+                        var result = ChangeVectorUtils.TryUpdateChangeVector(_connectionInfo.SourceTag, _connectionInfo.SourceDatabaseBase64Id, _lastDocumentEtag, current);
```

Comment:
If I understand correctly, we will only reach this point when `_lastDocumentEtag > lastEtag`, correct? (https://github.com/ravendb/ravendb/blob/v6.0/src/Raven.Server/Documents/Replication/Incoming/IncomingReplicationHandler.cs#L220). 
I'm concerned about conflicts or issues with sharding, but I haven't been able to identify any specific problematic scenarios so far, so I think it should be fine. :)

Reply:
@karmeli87  ?

Reply:
Note that this doesn't do a direct change vector merge, but only updates the change vector element of the replication counterpart.
We definitely will need to check that for sharding.
But I'm not sure if we have this code path and we dot allow output reduce to collection ATM.